nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -160,7 +160,17 @@ private static boolean[] getTypeParamsWithNullableUpperBound(
    */
   public static void checkGenericMethodCallTypeArguments(
       Tree tree, VisitorState state, NullAway analysis, Config config, Handler handler) {
-    List<? extends Tree> typeArguments = ((MethodInvocationTree) tree).getTypeArguments();
+    List<? extends Tree> typeArguments;
+    switch (tree.getKind()) {
+      case METHOD_INVOCATION:
+        typeArguments = ((MethodInvocationTree) tree).getTypeArguments();
+        break;
+      case NEW_CLASS:
+        typeArguments = ((NewClassTree) tree).getTypeArguments();
+        break;
+      default:
+        throw new RuntimeException("Unexpected tree kind: " + tree.getKind());
+    }
     if (typeArguments.isEmpty()) {
       return;
     }
```

Comment:
Below on line 178 don't you again need to handle the case where the tree.getKind() could be a NEW_CLASS?

Reply:
Good catch!  Will fix

Reply:
Now have new tests that exercise the rest of the method for the `NEW_CLASS` case